CMakeLists.txt

```diff
@@ -189,13 +201,18 @@ if(CONFIG_READABLE_ASM)
   zephyr_cc_option(-fno-partial-inlining)
 endif()
 
-zephyr_cc_option(-fno-asynchronous-unwind-tables)
-zephyr_cc_option(-fno-pie)
+if(NOT ${ZEPHYR_TOOLCHAIN_VARIANT} STREQUAL "xcc")
+  zephyr_cc_option(-fno-asynchronous-unwind-tables)
+  zephyr_cc_option(-fno-pie)
+endif()
+
 zephyr_cc_option(-fno-pic)
 zephyr_cc_option(-fno-strict-overflow)
 zephyr_cc_option(-Wno-pointer-sign)
 
-zephyr_compile_options_ifdef(CONFIG_STACK_CANARIES -fstack-protector-all)
+if(NOT ${ZEPHYR_TOOLCHAIN_VARIANT} STREQUAL "xcc")
+  zephyr_compile_options_ifdef(CONFIG_STACK_CANARIES -fstack-protector-all)
+endif()
```

Comment:
What the CMake code was originally expressing here was correct.

If the user has enabled STACK_CANARIES, but the toolchain does not support -fstack-protector-all, then the build system should not silently drop the protection.